docs/probe_hooking_notebook.md

```diff
@@ -0,0 +1,316 @@
+# Probes and Meters: Advanced Hooking Examples for Notebooks
```

Comment:
I like the idea of having some sort of hooking example in the docs. Ideally it could go in the same doc (or notebook?) and be a more advanced Example 3

Reply:
Tabling for now based on brief discussion of how to insert a `Probe` from a notebook without hooking, but redefining a function (e.g. `forward`) of a model or attack. The exact steps for this particular approach has not been determined yet.